src/RFSCTraceBuilder.cpp

```diff
@@ -182,10 +187,9 @@ void RFSCTraceBuilder::cancel_replay(){
    * prune all later decisions. */
   int blame = -1; /* -1: All executions lead to this failure */
   for (int i = 0; i <= prefix_idx; ++i) {
-    blame = std::max(blame, prefix[i].decision);
+    blame = std::max(blame, prefix[i].get_decision_depth());
   }
-  assert(int(decisions.size()) > blame);
-  decisions.resize(blame+1, decisions[0]);
+  prefix[blame].decision_ptr->prune_decisions();
```

Comment:
This is incorrect. The step numbers (prefix indices) do not correspond to the decision depths. You need to keep track of the index into prefix (or the decision node pointer), so you can prune the right node.